pool/pool.go

```diff
@@ -1343,6 +1351,12 @@ func (p *Pool) Dial(ctx context.Context) error {
 				if p.logger != nil {
 					p.logger.Warn("failed to dial client", zap.String("address", addr), zap.Error(err))
 				}
+
+				// In case of the pool has only one endpoint, we should return the error.
+				if len(clients) == 0 {
```

Comment:
in order to clarify problems also for > 1 endpoints, we can always return composite error `type DialError []error`. Then there won't be need to do this extra check. What do u think?

Reply:
It is possible, but we have to clearly understand that it was _wrong_ scheme error, we should skip this client. In another case try to call it, because they are just unhealthy and may be healthy later

Reply:
i'd check _all_ parameterized endpoints: they must follow https://pkg.go.dev/github.com/nspcc-dev/neofs-sdk-go/client#PrmDial.SetServerURI requirements (we should to write about it in `NewNodeParam` docs). If **any** endpoint has incorrect format, we immediately return error.

Summary:
```go
// Endpoint must be ... (similar to client)
func NewNodeParam(endpoint string) 

func NewPool(nodes []NodePrm) (*Pool, error)  {
  // input format check first
  for i := range nodes {
    if !isValid(nodes[i]) {
      return nil, fmt.Errorf("invalid endpoint #%d: %w", i, err) 
    } 
  }

  // then dial
  for i := rane nodes {
    // dial
  }
}
```

Reply:
Sounds good, I will update the PR